src/test/java/edu/cornell/kfs/sys/service/impl/AwsSecretServiceImplTest.java

```diff
@@ -0,0 +1,107 @@
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        awsSecretServiceImpl = null;
+    }
+
+    @Test
+    void testGetSingleStringValueFromAwsSecret() {
+        String actualSecretValue = awsSecretServiceImpl.getSingleStringValueFromAwsSecret(SINGLE_STRING_SECRET_KEY_NAME, false);
+        assertEquals(SINGLE_STRING_SECRET_VALUE, actualSecretValue);
+    }
+    
+    @Test
+    void testDateSetAndGet() throws ParseException {
+        Date date = new Date(Calendar.getInstance().getTimeInMillis());
+        awsSecretServiceImpl.updateSecretDate(SINGLE_DATE_SECRET_KEY_NAME, false, date);
+        
+        Date secretDate = awsSecretServiceImpl.getSingleDateValueFromAwsSecret(SINGLE_DATE_SECRET_KEY_NAME, false);
+        assertEquals(date.toString(), secretDate.toString());
+    }
+    
+    @Test
+    void testBooleanSetAndGet() {
+        boolean initialValue = awsSecretServiceImpl.getSingleBooleanFromAwsSecret(SINGLE_BOOLEAN_SECRET_KEY_NAME, false);
+        boolean expectedNewBoolean = !initialValue;
+        awsSecretServiceImpl.updateSecretBoolean(SINGLE_BOOLEAN_SECRET_KEY_NAME, false, expectedNewBoolean);
+        
+        boolean actualNewBoolean = awsSecretServiceImpl.getSingleBooleanFromAwsSecret(SINGLE_BOOLEAN_SECRET_KEY_NAME, false);
+        
+        assertEquals(expectedNewBoolean, actualNewBoolean);
+    }
+    
+    @Test
+    void testPojo() {
+        String newUniqueString = UUID.randomUUID().toString();
+        Date newDate = new Date(Calendar.getInstance().getTimeInMillis());
+        
+        AwsSecretePojoBasic pojo = awsSecretServiceImpl.getPojoFromAwsSecret(BASIC_POJO_SECRET_KEY_NAME, false, AwsSecretePojoBasic.class);
+        pojo.setChangeable_string(newUniqueString);
+        pojo.setUpdate_date(newDate);
+        boolean newBooleanTest = !pojo.isBoolean_test();
+        pojo.setBoolean_test(newBooleanTest);
+        awsSecretServiceImpl.updatePojo(BASIC_POJO_SECRET_KEY_NAME, false, pojo);
+        
+        AwsSecretePojoBasic pojoNew = awsSecretServiceImpl.getPojoFromAwsSecret(BASIC_POJO_SECRET_KEY_NAME, false, AwsSecretePojoBasic.class);
+        assertEquals(newUniqueString, pojoNew.getChangeable_string());
+        assertEquals(BASIC_POJO_STATIC_STRING_VALUE, pojoNew.getStatic_string());
+        assertEquals(BASIC_POJO_NUMBER_VALUE, pojoNew.getNumber_test());
+        assertEquals(newDate.toString(), pojoNew.getUpdate_date().toString());
+        assertEquals(newBooleanTest, pojo.isBoolean_test());
+    }
+    
+    @Test 
+    void testBuildFullAwsKeyNameAnyNamespace() {
+        String keyName = "foo";
+        String actualFullNameSpace = awsSecretServiceImpl.buildFullAwsKeyName(keyName, false);
+        String expectedFullNameSpace = KFS_SHARED_NAMESPACE + keyName;
+        assertEquals(expectedFullNameSpace, actualFullNameSpace);
+    }
+    
+    @Test 
+    void testBuildFullAwsKeyNameInstanctNamespace() {
```

Comment:
Minor typo in method name.

Also, this test class doesn't appear to call out to AWS with environment-specific keys. Should some dummy environment keys be used to test that out in some new test methods?

Reply:
I changed the name of the function.
Yes, I can add kfs/kfs-local namespace secrets to test with